src/utils/issue.ts

```diff
@@ -233,29 +264,55 @@ export function getOwnerRepoFromHtmlUrl(url: string) {
 
 export async function getAvailableOpenedPullRequests(context: Context, username: string) {
   const { reviewDelayTolerance } = context.config;
-  if (!reviewDelayTolerance) return [];
+  if (!reviewDelayTolerance) return { approved: [], changes: [] };
 
   const openedPullRequests = await getOpenedPullRequestsForUser(context, username);
-  const result: (typeof openedPullRequests)[number][] = [];
+  const approved = [] as unknown[];
+  const changes = [] as unknown[];
 
   for (let i = 0; openedPullRequests && i < openedPullRequests.length; i++) {
     const openedPullRequest = openedPullRequests[i];
     if (!openedPullRequest) continue;
     const { owner, repo } = getOwnerRepoFromHtmlUrl(openedPullRequest.html_url);
     const reviews = await getAllPullRequestReviews(context, openedPullRequest.number, owner, repo);
 
-    if (reviews.length > 0) {
-      const approvedReviews = reviews.find((review) => review.state === "APPROVED");
-      if (approvedReviews) {
-        result.push(openedPullRequest);
+    // Determine the latest review state
+    const latestReview = reviews[reviews.length - 1];
+    const latestReviewState = latestReview?.state;
+
+    if (latestReviewState === "APPROVED") {
+      approved.push(openedPullRequest);
+      continue;
+    }
+
+    if (latestReviewState === "CHANGES_REQUESTED") {
+      changes.push(openedPullRequest);
+
+      // Track the time of the last "CHANGES_REQUESTED"s
+      const lastChangesRequestedTime = latestReview.submitted_at ? new Date(latestReview.submitted_at).getTime() : null;
+
+      // Fetch timeline or comments to check if reviewer has been re-requested
+      const reviewRequests = await getReviewRequestsTimeline(context, openedPullRequest.number, owner, repo);
+
+      // Find if any review request was made after the last changes requested
+      const isReviewRequestedAfterChanges = lastChangesRequestedTime
+        ? reviewRequests.some((request) => new Date(request.created_at).getTime() > lastChangesRequestedTime)
+        : false;
+
+      // If reviewer was re-requested after changes were requested, mark as completed and increment approved count
+      if (isReviewRequestedAfterChanges) {
+        approved.push(openedPullRequest);
+        changes.pop();
+        continue;
       }
     }
 
-    if (reviews.length === 0 && new Date().getTime() - new Date(openedPullRequest.created_at).getTime() >= getTimeValue(reviewDelayTolerance)) {
-      result.push(openedPullRequest);
+    if (reviews.length === 0 && new Date().getTime() - new Date(openedPullRequest.created_at).getTime() >= getTimeValue("1m")) {
```

Comment:
why is it `1m`?

Reply:
was debugging and mistakenly missed it in the diffs